### PR TITLE
FEATURE: BB-1410 Subtotals menu

### PR DESCRIPTION
**Pull request by petrjanu**

### CODE REVIEW

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3536/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2830/ : SUCCESS in 2m 45s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1508/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3537/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/915/ : SUCCESS in 21m 07s

**Comment by petrjanu** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-852.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3552/ : SUCCESS in 36s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/920/ : FAILURE in 25m 40s

File: src/translations/en-US.json [hunk 1]
```diff
@@ -303,5 +303,15 @@
       "value": "Aggregations",
       "comment": "",
       "limit": 0
+   },
+   "visualizations.menu.aggregations.rows": {
+      "value": "Rows",
+      "comment": "Heading of row attributes list aggregation submenu of Table component. Selecting an attribute enables aggregation on selected attribute (adds 'total' or 'subtotal' row in attribute column of table).",
```
Comment (tomassatinsky) [inline src/translations/en-US.json]: I think we need to come up with something simpler for the translators. They are not so familiar with the domain. I think a comment like `"Row" here means table row` would do the job.

File: src/helpers/agGrid.ts [hunk 2]
```diff
@@ -321,10 +323,10 @@ export const getRowTotals = (
         columnKeys.filter((key: any) => {
             const currentKey = key.split(FIELD_SEPARATOR).pop();
             const fieldType = currentKey.split(ID_SEPARATOR)[0];
-            if (fieldType === 'a') {
+            if (fieldType === FIELD_TYPE_ATTRIBUTE) {
```
Comment (tomassatinsky) [inline src/helpers/agGrid.ts]: 👍

File: src/components/core/pivotTable/AggregationsMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        columnTotals: IColumnTotal[],
+        measureLocalIdentifiers: string[],
+        rowAttributeHeaders: Execution.IAttributeHeader[]
+    ) {
+        const { intl, onAggregationSelect, showSubmenu } = this.props;
+        const firstAttributeIdentifier = this.getFirstAttributeIdentifier(rowAttributeHeaders);
+
+        return AVAILABLE_TOTALS.map((totalType: AFM.TotalType) => {
+            const isSelected = menuHelper.isTotalEnabledForAttribute(firstAttributeIdentifier, totalType, columnTotals);
+            const onClick = this.onItemClickFactory(
```
Comment (tomassatinsky) [inline src/components/core/pivotTable/AggregationsMenu.tsx]: would the following code work as well?
```
const onClick = () => this.props.onAggregationSelect({
            type: totalType,
            measureIdentifiers,
            include: !isSelected,
            attributeIdentifier: rowAttributeHeaders[0].attributeHeader.localIdentifier
        });
```

Comment (tomassatinsky) [reply src/components/core/pivotTable/AggregationsMenu.tsx]: It might be a bit simpler

**Comment by petrjanu** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-852.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

File: examples/test/PivotTableMenu_test.js [hunk 4]
```diff
@@ -32,7 +32,7 @@ const clickOnMenuAggregationItem = async (t, cell, aggregationItemClass) => {
     await t.hover(cell);
     await t.click(menu);
 
-    const sumTotal = Selector(aggregationItemClass);
+    const sumTotal = Selector(aggregationItemClass).find('.gd-aggregation-menu-item-inner');
```
Comment (tomassatinsky) [inline examples/test/PivotTableMenu_test.js]: Seems like use just the `s-` classes here

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3559/ : SUCCESS in 23s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2847/ : FAILURE in 12m 21s

File: src/components/core/pivotTable/AggregationsSubMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+                    <div className="s-table-header-submenu-content">
+                        <Header>{intl.formatMessage({ id: 'visualizations.menu.aggregations.rows' })}</Header>
+                        {this.renderSubMenuItems()}
+                    </div>
+                </ItemsWrapper>
+            </SubMenu>
+        );
+    }
+
+    private getAttributeLocalIdentifier(
```
Comment (tomassatinsky) [inline src/components/core/pivotTable/AggregationsSubMenu.tsx]: Could you move these two function out of the file? It hurts to see digging into execution result from within a React component.

Comment (tomassatinsky) [reply src/components/core/pivotTable/AggregationsSubMenu.tsx]: it can be re-used in `getFirstAttributeIdentifier` and `itemClickFactory`

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3560/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/924/ : SUCCESS in 25m 06s

**Comment by petrjanu** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-852.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3562/ : SUCCESS in 29s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2850/ : SUCCESS in 3m 18s

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3563/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/925/ : FAILURE in 24m 03s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/780/ : SUCCESS in 31s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2855/ : SUCCESS in 2m 36s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1519/ : UNSTABLE in 15m 19s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/766/ : SUCCESS in 18m 22s

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1522/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3570/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/927/ : UNSTABLE in 11m 32s

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3571/ : SUCCESS in 27s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2859/ : SUCCESS in 3m 45s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1524/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3572/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/928/ : UNSTABLE in 15m 38s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3573/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2860/ : SUCCESS in 4m 14s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/783/ : FAILURE in 48s
- gooddata-react-components-unit-tests-zuul-docker-el7 : SKIPPED
- gooddata-react-components-storybook-zuul-docker : SKIPPED
- gooddata-react-components-examples-zuul-docker : SKIPPED

**Comment by martinmosko** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/784/ : SUCCESS in 47s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2861/ : SUCCESS in 5m 46s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1525/ : SUCCESS in 17m 46s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/769/ : SUCCESS in 18m 47s

**Pull request merged**